neurons/miner.py

```diff
@@ -1,175 +1,185 @@
-
+import asyncio
+import copy
 import time
-import typing
+
 import bittensor as bt
-import datetime as dt
-import os
-import datetime
-import asyncio
+from dotenv import load_dotenv
+from huggingface_hub import HfApi, login as hf_login
+import huggingface_hub
+import onnx
 import cancer_ai
+import typing
+import argparse
+
+from cancer_ai.validator.utils import run_command
+from cancer_ai.validator.model_run_manager import ModelRunManager, ModelInfo
+from cancer_ai.validator.dataset_manager import DatasetManager
+from cancer_ai.validator.competition_manager import COMPETITION_HANDLER_MAPPING
 
-# import base miner class which takes care of most of the boilerplate
-from cancer_ai.base.miner import BaseMinerNeuron
+from cancer_ai.base.miner import BaseNeuron
 from cancer_ai.chain_models_store import ChainMinerModel, ChainModelMetadataStore
+from cancer_ai.utils.config import path_config, add_miner_args
 
 
-class Miner(BaseMinerNeuron):
-    """
-    Your miner neuron class. You should use this class to define your miner's behavior. In particular, you should replace the forward function with your own logic. You may also want to override the blacklist and priority functions according to your needs.
+class MinerManagerCLI:
+    def __init__(self, config=None):
 
-    This class inherits from the BaseMinerNeuron class, which in turn inherits from BaseNeuron. The BaseNeuron class takes care of routine tasks such as setting up wallet, subtensor, metagraph, logging directory, parsing config, etc. You can override any of the methods in BaseNeuron if you need to customize the behavior.
+        # setting basic Bittensor objects
+        base_config = copy.deepcopy(config or BaseNeuron.config())
+        self.config = path_config(self)
+        self.config.merge(base_config)
+        BaseNeuron.check_config(self.config)
+        bt.logging.set_config(config=self.config.logging)
+        bt.logging.info(self.config)
+
+    @classmethod
+    def add_args(cls, parser: argparse.ArgumentParser):
+        """Method for injecting miner arguments to the parser."""
+        add_miner_args(cls, parser)
+
+    async def upload_to_hf(self) -> None:
+        """Uploads model and code to Hugging Face."""
+        bt.logging.info("Uploading model to Hugging Face.")
+        hf_api = HfApi()
```

Comment:
Are we going to use it in multiple places? just a gut feeling to make it self.hf_api in init

Reply:
I moved it from there. it's not used anywhere else